design/deletion-plugins.md

```diff
@@ -0,0 +1,80 @@
+# Deletion Plugins
```

Comment:
I'm currently leaning towards doing just `DeleteItemActions` right now, and deferring the `DeleteActions` plugins until later, if at all. For @carlisia and @ashish-amarnath, would you prefer I remove this file?

Reply:
I would say we keep is and list it as alternatives considered.